### PR TITLE
JAX-RS body readers for Vert.x JsonObject and JsonArray

### PR DESCRIPTION
**Pull request by tsegismont**
This allows to use Vert.x JsonObject and JsonArray as JAX-RS request content.

As many Vert.x client APIs take json content (e.g. Mongo client), this might be handy.

### CODE REVIEW

**Comment by tsegismont** [comment]
@FroMage @cescoffier can one of you please review? Thanks.

This follows-up on #1549

File: integration-tests/vertx/pom.xml
Comment (gsmet) [inline integration-tests/vertx/pom.xml]: Undertow comes with the resteasy extension (and it shouldn't be provided anyway).

**Review by gsmet — CHANGES_REQUESTED** [review]
Looks like an interesting addition. There's an issue with the dependencies though.

I'll let other people review it in more details, I just made a quick scan.

**Review by FroMage — APPROVED** [review]
LGTM, provided the `provided` issue raised by @gsmet is fixed.

File: integration-tests/vertx/pom.xml
Comment (tsegismont) [reply integration-tests/vertx/pom.xml]: Not sure what you mean. Without it the tests did not run.

**Comment by tsegismont** [comment]
@gsmet @FroMage thank you both for reviewing.

@FroMage do you mean it's been fixed in master and I only need to rebase the PR?

File: integration-tests/vertx/pom.xml
Comment (gsmet) [reply integration-tests/vertx/pom.xml]: Just to be clear: this dependency should be dropped.

Comment (gsmet) [reply integration-tests/vertx/pom.xml]: Let me take a look at it.

Comment (gsmet) [reply integration-tests/vertx/pom.xml]: I confirm it works without it with your PR on top of master and a clean build.

It should definitely be removed.

Comment (tsegismont) [reply integration-tests/vertx/pom.xml]: Just rebased and updated.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Let's wait for CI and merge.

**Comment by gsmet** [comment]
And merged! Thanks!

**Pull request merged**
